### DISCUSSION TITLE
Install Guide for beginners in Linux Mint to get work Fanatec CSL Elite loadcell pedals connected with USB to PC

### DISCUSSION THREAD

**Discussion by SarahLaHonda** [gotzl/hid-fanatecff#90] [Q&A]
Hello all
i am totally new in Linux. I have a Thrustmaster T300 and this works with the guide from the tm-kernel-Autor. But now, i dont know, how i get the installation to work. My Pedals are the Fanatec CSL Elite loadcell V1 ( not the new release ) and they are connected with the Fanatec USB adapter ( 3 pedals ) to my PC. Can Someone help me pls?

**Comment by gotzl** [comment]
Hi,
I have the Fanatec CLS Elite too, and I use its USB port to connect it directly to the PC, w/o needing the adapter. Is this maybe an option for you as well?

**Reply by SarahLaHonda** [reply]
For using with Loadcell, the USB-adapter is mandatory

**Reply by SarahLaHonda** [reply]
throttle and clutch working, but brake Load-cell not

**Reply by SarahLaHonda** [reply]
ah no no. i mean the adapater for 3 pedals ( throttle, Loadcell-brake, normal brake for Clutch ). Not the normal one for 2 pedals.

**Reply by SarahLaHonda** [reply]
in evtest nothing happens when i push the loadcell

**Reply by gotzl** [reply]
Ok, seems Fanatec changed s.t. here, as I said I've the same pedals w/ load-cell kit. But the pedals them-self have a USB port...
The problem with the adapter is that it has different modes... And I suppose yours is not already in 'pedals' mode? Can you run `journalctl -xfk`, then plug the adapter in and show what's logged?

> in evtest nothing happens when i push the loadcell

The other pedals work? Just the loadcell doesn't work?

**Reply by SarahLaHonda** [reply]
Jul 22 18:01:57 sarah-desktop kernel: usb 1-3: new full-speed USB device number 14 using xhci_hcd
Jul 22 18:01:57 sarah-desktop kernel: usb 1-3: New USB device found, idVendor=0eb7, idProduct=6204, bcdDevice= 0.01
Jul 22 18:01:57 sarah-desktop kernel: usb 1-3: New USB device strings: Mfr=1, Product=2, SerialNumber=0
Jul 22 18:01:57 sarah-desktop kernel: usb 1-3: Product: FANATEC CSL Elite Pedals LC
Jul 22 18:01:57 sarah-desktop kernel: usb 1-3: Manufacturer: FANATEC
Jul 22 18:01:57 sarah-desktop kernel: input: FANATEC FANATEC CSL Elite Pedals LC as /devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3:1.0/0003:0EB7:6204.000D/input/input37
Jul 22 18:01:57 sarah-desktop kernel: fanatec 0003:0EB7:6204.000D: input,hidraw1: USB HID v1.11 Joystick [FANATEC FANATEC CSL Elite Pedals LC] on usb-0000:00:14.0-3/input0

**Reply by SarahLaHonda** [reply]
> The other pedals work? Just the loadcell doesn't work?

yes. Throttle and clutch works fine. tested in evtest and Assetto Corsa Competizione

**Reply by SarahLaHonda** [reply]
Evtest

Supported events:
  Event type 0 (EV_SYN)
  Event type 1 (EV_KEY)
    Event code 336 (BTN_GEAR_DOWN)
  Event type 3 (EV_ABS)
    Event code 0 (ABS_X)
      Value      6
      Min        0
      Max     4095
      Fuzz      15
      Flat     255
    Event code 1 (ABS_Y)
      Value  65535
      Min        0
      Max    65535
      Fuzz     255
      Flat    4095
    Event code 2 (ABS_Z)
      Value      0
      Min        0
      Max     4095
      Fuzz      15
      Flat     255
    Event code 3 (ABS_RX)
      Value      0
      Min        0
      Max     4095
      Fuzz      15
      Flat     255
Properties:
Testing ... (interrupt to exit)

here only ABS X ( throttle ) and ABS Z (clutch) works by pressing the pedal. 

on Windows the loadcell works normal ( same Computer )

**Reply by gotzl** [reply]
Ah, great, then the adapter is already in the correct mode! And I see you've also managed to install the driver... Hmm... have to think about what's missing then ...

**Reply by gotzl** [reply]
Just a quick thing, I might have missunderstood when you were talking about USB adapter. You did not talk about [this one](https://fanatec.com/eu-de/accessories/pedal-accessories/clubsport-usb-adapter), but the [this one](https://forum.fanatec.com/uploads/762/36S5W77PS9PN.jpeg), right?

In this case, we really do have the same setup... Which makes it even more puzzling why it doesn't work for you...

**Reply by SarahLaHonda** [reply]
the second one right.

**Reply by SynthetikzZ** [reply]
To my understanding the load cell brake acts exactly like the normal potentiometer brake if wired correctly and calibration is done on the pedal itself if I remember correctly. So the loadcell should basically work as any other pedal. Maybe try to plug the brake into the throttle and the throttle into the brake and test again



Edit: browsing trough fanatec forums it seems like you have to plugin the LC directly via USB to the PC, not into the adapter. 


My csl throttle has an "loadcell" input port on its own so idk.

**Reply by gotzl** [reply]
It also seam that the udev-rule didn't run. Maybe `evdev-joystick --evdev /dev/input/by-id/usb-FANATEC_FANATEC_CSL_Elite_Pedals_LC-event-* --deadzone 0 --fuzz 0 --flatness 0` changes s.t.?

**Reply by SarahLaHonda** [reply]
> It also seam that the udev-rule didn't run. 

how i start it or create it, manually possible?

**Reply by gotzl** [reply]
> > It also seam that the udev-rule didn't run.
> 
> how i start it or create it, manually possible?

See [here](https://github.com/gotzl/hid-fanatecff?tab=readme-ov-file#compile-and-install-the-driver), but I don't think it'll help. It'll also just execute the command I showed ...

> Edit: browsing trough fanatec forums it seems like you have to plugin the LC directly via USB to the PC, not into the adapter.

The LC load-cell kit brings its own kind of 'USB adapter' where you've to plug in all the pedals, it looks [like this](https://forum.fanatec.com/uploads/762/36S5W77PS9PN.jpeg). The USB PID is also the correct one, and since the pedal is working in Windows, the wiring seems to be alright as well....
The test SynthetikzZ suggests would be interesting, though..

**Reply by SarahLaHonda** [reply]
i tried everything.  the LC didnt work. I follow your work here for the future. i think for simracing and eurotruck/Americantruck simulator i use my dual-boot with windows11 and all other using Linux to learn as much as possible and have a great OS experience

**Reply by gotzl** [reply]
If you want to dig a little further, you could check with wireshark if there is really data coming from the LC. For this you've to start wireshark with sudo, then select `usbmon1` (to match the `usb1` in the kernel log when connecting, ie in your case `/devices/pci0000:00/0000:00:14.0/usb1/...`), then start capturing and see if data is coming in once you press the brake.

**Reply by SarahLaHonda** [reply]
it  works now ;) yeah, challenge solved. but now, how can i set the sensitivity from the lc? its to soft, I would like to need more strength for 100%

**Reply by SynthetikzZ** [reply]
> it works now ;) yeah, challenge solved. but now, how can i set the sensitivity from the lc? its to soft, I would like to need more strength for 100%

Trough the Tuning menu on your wheelrim normally with an fantec wheelbase. Dual boot into windows and change it there once is an option.

**Reply by SarahLaHonda** [reply]
> Trough the Tuning menu on your wheelrim normally with an fantec wheelbase.<- have only pedals from fana, wheelbase ist Thrustmaster | Dual boot into windows and change it there once is an option. <- thx

**Reply by SarahLaHonda** [reply]
> > it works now ;) yeah, challenge solved. but now, how can i set the sensitivity from the lc? its to soft, I would like to need more strength for 100%
> 
> Trough the Tuning menu on your wheelrim normally with an fantec wheelbase. Dual boot into windows and change it there once is an option.

doesnt works

**Reply by SarahLaHonda** [reply]
tried 3 times. set new min/max and set to 80% or 100%. then reboot to Linux. Low pressure like everytime, look back in windows. lc level is 40%. changed again, reboot, low pressure. reboot to win, 40% again.

**Reply by gotzl** [reply]
The driver applies the 40% per default on load, so it overwrites the setting you've set in windoes ;)

You can change the value in Linux by writing 0 to 9 (?) to the `load` sysfs file. St. like `echo 6 > /sys/module/hid_fanatec/drivers/hid:fanatec/0003:0EB7:6204.*/load`, not completely sure this path is right for you.
If you want to change the default, you can create a file like `/etc/modprobe.d/ftec-load.conf` with content `options hid_fanatec init_load=6` (not tested, but s.t. like that should work ..)

> it works now ;) yeah, challenge solved.

Just out of curiosity, what did the trick?

Edit: Also, I just found I don't get input from the pedals anymore after `rmmod hid-fanatec` followed by `modprobe hid-fanatec`, unless I replug the USB.

**Comment by SarahLaHonda** [comment]
Wenn ich das richtig gelesen habe, sind alle beteiligten aus Deutschland?! Ich bin völlig überwältigt von eurer Hilfsbereitschaft. Vielen vielen Dank. Alles läuft soweit, nur die Performance von Assetto Corsa Competizione ist miserabel. im Vergleich, auf Win11 schaff ich 2160p60fps ( DLSS Leistung) Auf Mint hab ich schon probs bei 1440p. 2160p geht gar nicht. So bleibt es bei ETS2/ATS und ACC auf Win11.

Nochmal dickes Danke

**Reply by SynthetikzZ** [reply]
Bei mir läuft acc exakt genauso wie auf Windows aber ich hab auch eine amd karte, nividia und Linux sind halt nochmal ein ganz anderes Thema :D 

Ich zock aber eigentlich nurnoch rfactor2, das force feedback ist Weltklasse und von diesem Treiber hier komplett unterstützt. 

Übrigens dualbooting kann schonmal für Probleme sorgen, Windows Updates können dir dein Linux zerschießen.  Wenn du dauerhaft auf Linux umsteigen willst würde Ich das Windows komplett runter schmeißen oder zur Not eine VM aufsetzen. (Geht auch mit GPU passtrough, mit paar Skripts auch wenn man nur eine einzelne GPU im System hat) . Dann kann dir das windows auch nicht dein Linux zerschießen.

**Reply by gotzl** [reply]
Ich habe keine Erfahrung mit Nvidia, aber ein Performance-Drop gegenueber Windows sollte auch damit nicht sooo gross sein... Vlt sind die Power Einstellungen in Linux nicht richtig und die Graka wird gedrosselt?

**Reply by irseny** [reply]
Die Performance von ACC geht für ich normalerweise auch in Ordnung. Da aber die Frametimes merkbar instabil waren, versetze ich die Grafikkarte meistens in high perfomance Modi.  Macht in meinem Fall viel aus. Ich hab mit CoreCtrl gute Erfahrungen gemacht. Über die nvidia-settings ließ sich, so meine ich, auch ein Performance Modus aktivieren. Oder man kennt die entsprechenden Sysfs-Files.
